lib/truncato/truncato.rb

```diff
@@ -26,26 +26,14 @@ def self.truncate source, user_options={}
   private
 
   def self.truncate_html source, options
+    source = source.encoding == Encoding::UTF_8 ? source.unicode_normalize : source
     self.do_truncate_html(source, options) ? self.do_truncate_html(with_artificial_root(source), options) : nil
   end
 
   def self.do_truncate_html source, options
-    begin
-      source = source.unicode_normalize
-    rescue Encoding::CompatibilityError
-      # Only Unicode encodings can be normalized.
-      #
-      # Ruby docs:
-      # > In this context, 'Unicode Encoding' means any of UTF-8,
-      # > UTF-16BE/LE, and UTF-32BE/LE, as well as GB18030, UCS_2BE, and
-      # > UCS_4BE. Anything else than UTF-8 is implemented by converting
-      # > to UTF-8, which makes it slower than UTF-8.
-    end
-
     truncated_sax_document = TruncatedSaxDocument.new(options)
 
-    # Only nokogiri >= 1.17 accept Encoding object, older needs a String as encoding
-    parser = Nokogiri::HTML::SAX::Parser.new(truncated_sax_document, source.encoding.to_s)
+    parser = Nokogiri::HTML::SAX::Parser.new(truncated_sax_document, source.encoding)
```

Comment:
This will need a change in the gemspec, since users could very well use the `truncato` gem with an older `nokogiri`:

```diff
-   spec.add_dependency "nokogiri", ">= 1.7.0", "<= 2.0"
+   spec.add_dependency "nokogiri", "~> 1.17.0"
```